koku/sources/sources_error_message.py

```diff
@@ -68,12 +76,14 @@ def _extract_from_validation_error(self):
         if isinstance(self._error, ValidationError):
             err_dict = self._error.detail
             LOG.info(f"validation error: {self._error}. Validation detail {err_dict}")
-            err_key = list(err_dict.keys()).pop()
-            err_body = err_dict.get(err_key, []).pop()
-            if err_body:
-                err_msg = err_body.encode().decode("UTF-8")
-            else:
+            err_key = list(err_dict).pop()
+            try:
+                err_body = err_dict.get(err_key, []).pop()
+            except (TypeError, IndexError):
```

Comment:
What raises a TypeError here?

Also, we can avoid the IndexError by giving the default list a value, such as `[None]`.
```suggestion
            err_key = list(err_dict).pop()
            err_body = err_dict.get(err_key, [None]).pop()
            if err_body:
                err_msg = err_body.encode().decode("UTF-8")
```

Reply:
`{}.pop()` raises a `TypeError`. This was the source of the original error. We assumed the data would always be a list and (because of another bug) we're getting a dict instead.

> Also, we can avoid the IndexError by giving the default list a value, such as [None]

Yup, that would work.

Reply:
We still need to catch and handle the `TypeError`, though.

Reply:
Any idea what the `.encode.decode('uff-8')` here accomplishes?